lib/google-apis/auth-client.js

```diff
@@ -15,136 +15,57 @@
- *  exponential retry backoff.
- * @property {GoogleAuthClient} _authClient - an authorized GoogleAuth client
- *  instance
- * @property {Boolean} _hasInited - Indicator denoting whether or not the
- *  underlying GoogleAuth client has been authorized and is ready to interact
- *  with the service
- * @property {Error|Null} _initError - Will hold the error given by the
- *  underlying client if applicable during initialization.
- * @property {Boolean} _isRequesting - Indicator denoting whether or not the
- *  client is currently making a request against the service
- * @property {Array} _requestQueue - The pending request queue, requests that
- *  will be executed against the service
- * @property {GoogleAuth} _authFactory - The GoogleAuth factory for creating
- *  authorized clients
- * @property {String} _key - An optional key used for simple authentication
- * @property {String} _projectId - The project Id used to request against the
- *  service
- * @property {Boolean} _shouldReportErrors - Indicator denoting whether or not
- *  the client should actually attempt requests against the service
+ * @classdesc The RequestHandler class provides a centralized way of managing a
+ * pool of ongoing requests and routing there callback execution to the right
+ * handlers. The RequestHandler relies on the diag-common request factory
+ * and therefore only manages the routing of execution to the proper callback
+ * and does not do any queueing/batching. The RequestHandler instance has
+ * several properties: the projectId property is used to create a correct url 
+ * for interacting with the API and key property can be optionally provided a 
+ * value which can be used in place of default application authentication. The 
+ * shouldReportErrors property will dictate whether or not the handler instance 
+ * will attempt to send payloads to the API. If it is false the handler will
+ * immediately call back to the completion callback with a constant error value.
+ * @property {Function} _requestFactory - a authorized request instance that
+ *  provides the transport layer for requesting against the Error Reporting API
+ * @property {String} _projectId - the project id used to uniquely identify and
+ *  address the correct project in the Error Reporting API
+ * @property {String|Null} _key - an optional API key for use with the service
+ *  which can be used in place of default application authentication
  */
-function AuthClient(projectId, shouldReportErrors, key) {
-
-  this._authClient = null;
-  this._hasInited = false;
-  this._initError = null;
-  this._isRequesting = false;
-  this._requestQueue = [];
-  this._authFactory = new GoogleAuth();
-
-  this._key = isString(key) ? key : null;
+function RequestHandler(projectId, shouldReportErrors, key) {
+  
+  this._requestFactory = commonDiag.utils.authorizedRequestFactory([
+    'https://www.googleapis.com/auth/cloud-platform'
```

Comment:
@steren: the API scope seems really broad. Any reason why error reporting needs full `cloud-platform` scope? Could there be a finer grained scope we could use here?